### PR TITLE
Release instructions

### PR DESCRIPTION
**Pull request by muffato**
TOLIT-1870

I've added a document about how to make a release. It piggies back on the nf-core one, with the additions of:

- Naming the release
- Getting the Zenodo record right
- Getting an upload onto WorkflowHub

Since it's just 1 extra page on top of the previous PR, I've updated the dev website already: http://pipelines.tol-dev.sanger.ac.uk/docs/contributing/pipeline-release

### CODE REVIEW

**Review by BethYates — COMMENTED** [review]
Should there be a blank space between creativity and ! on line 30?

**Comment by muffato** [comment]
I don't mind either way. I'd just want the rule we agree to be consistently used. The other occurrences of an exclamation mark in ToL documents are:

```
markdown/usage/filesystem.md:> Page in progress !
markdown/events/2023/meetup_nextflow_in_action.md:Tree of Life is the latest scientific programme of the Wellcome Sanger Institute, aiming at producing reference genome assemblies for tens of thousands of species. We decided early on that Nextflow would be the workflow manager of choice - what an adventure it has been ! In this presentation, I will tell you about our journey, starting from training users, setting up development environments, right up to advertising our complete pipelines and reaching out to users outside of the department, with some insights into deploying and automating the pipelines.
markdown/developers/modules.md:To simplify maintenance of the script, the module, and the pipeline, we recommend implementing in each script a way of printing a short usage message and a version number. This will also help you remember what the script is for and what arguments to pass !
markdown/resources/genome_after_party.md:_Only if there is an actual demand ! Use the form below to indicate interest._
```

All with a whitespace – maybe my fault, TBH. [My excuse is that I'm French :)](https://english.stackexchange.com/questions/4645/is-it-ever-correct-to-have-a-space-before-a-question-or-exclamation-mark#comment6964_4645)

> It's a common typo by French (because in French, there is a white space before a question/exclamation mark)

Whereas all the other documents (all under `markdown/events/` as far as I can tell) have _no_ whitespace before `!`.

I'll remove the whitespace

**Review by BethYates — APPROVED** [review]

File: markdown/developers/pipeline-release.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+- `dev` is the staging branch. Code on the `dev` branch should always pass tests and be functional (this is a condition of our [review guidelines](/docs/contributing/review_checklist)). Releasing a pipeline involves merging `dev` into `main`.
+- Feature branches are where development happens.
+
+## Release steps
+
+Follow the "Before you release" and "Steps to release" from the [nf-core release checklist](https://nf-co.re/docs/contributing/release_checklist) with the following adaptations:
+
+1. Replace "nf-core" with "sanger-tol".
+2. Since we develop directly off the sanger-tol repository, the version bump happens there, not "on your fork".
+3. The "core team member" who can activate the integration with Zenodo is @muffato.
```
Comment (gq1) [inline markdown/developers/pipeline-release.md:24]:
```suggestion
3. The "core team member" who can activate the integration with Zenodo is [@muffato](https://github.com/muffato).
```

Comment (gq1) [reply markdown/developers/pipeline-release.md:24]: Add @DLBPointon here as well?

File: markdown/developers/pipeline-release.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```text
+## [[${RELEASE_NUMBER}](https://github.com/sanger-tol/${PIPELINE_NAME}/releases/tag/${RELEASE_NUMBER})] - ${RELEASE_NAME} - [${RELEASE_DATE}]
+
+*Summary of the release*
+```
+
+## Integration with Zenodo
+
+Once you've made a release, a record is automatically created on Zenodo.
+Tell @muffato or @DLBPointon who can then do the following:
```
Comment (gq1) [inline markdown/developers/pipeline-release.md:77]:
```suggestion
Tell [@muffato](https://github.com/muffato) or [@DLBPointon](https://github.com/DLBPointon) who can then do the following:
```

File: markdown/developers/pipeline-release.md (hunk 1 above)
Comment (muffato) [reply markdown/developers/pipeline-release.md:24]: For now, I want it to be just me in the docs.
cf TOLIT-1871 the problem we have is that the GitHub integrations are all owned by different people. In that ticket I'll try to find a solution (I hope something along the lines of having a tolsoft account on Zenodo and asking the Zenodo helpdesk to move all integrations over to tolsoft)

**Review by gq1 — APPROVED** [review]

**Pull request merged**